official/resnet/resnet_run_loop.py

```diff
@@ -355,21 +370,35 @@ def resnet_main(flags, model_function, input_function, shape=None):
       allow_soft_placement=True)
 
   # Set up a RunConfig to save checkpoint and set session config.
-  run_config = tf.estimator.RunConfig().replace(save_checkpoints_secs=1e9,
-                                                session_config=session_config)
+  if not flags.use_distribution_strategy:
+    run_config = tf.estimator.RunConfig().replace(
+        save_checkpoints_secs=1e9, session_config=session_config)
+  else:
+    if flags.gpus_for_distribution_strategy == 0:
+      distribution = one_device_strategy.OneDeviceStrategy(
+          'device:CPU:0')
+    elif flags.gpus_for_distribution_strategy == 1:
+      distribution = one_device_strategy.OneDeviceStrategy(
+          'device:GPU:0')
+    else:
+      distribution = mirrored_strategy.MirroredStrategy(
+          num_gpus=flags.gpus_for_distribution_strategy
+      )
+    run_config = tf.estimator.RunConfig(distribute=distribution).replace(
+        save_checkpoints_secs=1e9, session_config=session_config)
```

Comment:
@guptapriya What is the effect of passing inter_ and intra_op parallelism with a dist strategy? Are they still respected?

Reply:
Hm, I am not a 100% sure. We did test with setting those as env variables during performance tuning but did not find significant benefits. But I don't believe we changed anything in dist strategy itself to support/not support them explicitly. @isaprykin perhaps can shed more light?

Reply:
There is currently an issue with TensorFlow not respecting thread limits (https://github.com/tensorflow/tensorflow/issues/18300), so we'll have to wait until that is resolved to test this.

Reply:
Hi!  We didn't change anything regarding inter/intra op options.  We indeed tried them out, but they weren't super impactful that time.  I also remember Reed saying that these options have reasonable defaults.